### PR TITLE
Fix issue 33356: Document integrity properly

### LINKED ISSUE #33356: Subresource Integrity check on Request instance with integrity property
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity

### What specific section or headline is this issue about?

How browsers handle Subresource Integrity

### What information was incorrect, unhelpful, or incomplete?

Checking the article for the [integrity property](https://developer.mozilla.org/en-US/docs/Web/API/Request/integrity) in the Request interface, it links to the article for Subresource Integrity. 

While reading that article, I was left wondering if, when using the Fetch api, passing a Request instance with an integrity value would be validated by the browser, just as if the property was set on the attribute of a script tag.

### What did you expect to see?

If this check also applied to requests that have an integrity check set in JavaScript, through `Request`'s constructor.

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/security/subresource_integrity`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/security/subresource_integrity/index.md
* Last commit: https://github.com/mdn/content/commit/864e7a830d3f658b4cd87098428097bda5ee5b84
* Document last modified: 2024-01-02T08:34:12.000Z

</details>

### PR DESCRIPTION
**Pull request by wbamberg**
Fixes https://github.com/mdn/content/issues/33356.

### CODE REVIEW

**Comment by Josh-Cena** [comment]
I'm not sure if this page is the right place to document it as it's a simple reflector of the internal state of the request. It should probably be on the `Request()` and/or `fetch()` page (before we merge them, that is), and a brief mention can be put here.

**Comment by wbamberg** [comment]
> I'm not sure if this page is the right place to document it as it's a simple reflector of the internal state of the request. It should probably be on the `Request()` and/or `fetch()` page (before we merge them, that is), and a brief mention can be put here.

I did think about this, and the reason I went with here is that we have a whole page to talk about it here, and it's nice to be able to use that space. If we document it in the constructor or `fetch()` (or even `RequestInit`) we'd only have one page to cover all the options, and it would get huge. We do something similar in (for example) https://developer.mozilla.org/en-US/docs/Web/API/Request/cache.

Alternative I suppose would be to create subpages for `RequestInit`, although we usually don't, for dictionaries.

**Comment by Josh-Cena** [comment]
We can either make a tree for `RequestInit`, or alternatively just have `fetch/integrity`, `fetch/cache`, etc. Not easy to design 😅

I don't have strong opinions against this either. The WebAPI information is really all over the place (especially HTML element stuff).

**Comment by wbamberg** [comment]
I think the best alternative is inline in RequestInit, and I'm happy to try that when it exists.

File: files/en-us/web/api/request/integrity/index.md [hunk 1]
```diff
@@ -10,30 +10,47 @@ browser-compat: api.Request.integrity
...
-console.log(myRequest.integrity); // "sha256-abc123"
+
+async function get(request) {
+  const response = await fetch(request);
+  console.log(response.status);
+  // ...
+}
 ```
 
+When the request is made, the browser computes the SHA-384 {{glossary("Cryptographic_hash_function", "hash")}} of the fetched resource, and if the result does not match the value specified in `integrity`, the browser will reject the fetch request.
```
Comment (estelle) [inline files/en-us/web/api/request/integrity/index.md:52]:
```suggestion
When the request is made, the browser computes the SHA-384 {{glossary("Cryptographic_hash_function", "hash")}} of the fetched resource. If the result does not match the value specified in `integrity`, the browser will reject the fetch request.
```
```suggestion
When the request is made, the browser computes the SHA-384 {{glossary("Cryptographic_hash_function", "hash")}} of the fetched resource, and if the result does not match the value specified in `integrity`, the browser will reject the fetch request.
```

**Review by estelle — APPROVED** [review]
Thanks! 🎉

**Comment by Josh-Cena** [comment]
@wbamberg Do you want to merge this or to move the content to `RequestInit` instead?

**Comment by wbamberg** [comment]
I put all the relevant stuff in the other PR: https://pr34437.content.dev.mdn.mozit.cloud/en-US/docs/Web/API/RequestInit#integrity so think we can close this PR without merging.

**Pull request closed without merging**